pkg/apis/vsphere/v1alpha1/cloud/types.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package cloud contains API types for the vSphere cloud provider.
+package cloud
+
+// Config is the vSphere cloud provider's configuration.
+type Config struct {
```

Comment:
this probably already been discussed: what is the plan to keep in-sync with cloud provider types? 
is vendor-in types from cloud provider repo in the long term plan? thanks

Reply:
That's correct. On both counts. Eventually these types will move into the CCM codebase.

Reply:
Can you open an issue in the CCM repo for this?